src/test/resources/Dockerfile

```diff
@@ -0,0 +1,32 @@
+FROM ubuntu:noble
```

Comment:
This PR introduced technical debt, as described in https://github.com/jenkinsci/workflow-api-plugin/pull/414#pullrequestreview-3010599259.

Reply:
Some, though not so much in this case: to update the Java version here you would also need to adjust `JDK` installation paths in the test, which you would do as part of the same PR that replaces the Ubuntu package name.